src/replay/svm_gateway.zig

```diff
@@ -92,21 +91,26 @@ pub const SvmGateway = struct {
 
     pub fn init(
         allocator: Allocator,
-        batch: []const ResolvedTransaction,
+        batches: []const replay.resolve_lookup.ResolvedBatch,
         params: Params,
     ) !SvmGateway {
         const zone = tracy.Zone.init(@src(), .{ .name = "SvmGateway.init" });
         defer zone.deinit();
 
-        var accounts = try BatchAccountCache.initSufficientCapacity(allocator, batch);
-        for (batch) |transaction| {
-            try accounts.load(
+        var max_accounts: usize = 0;
+        for (batches) |batch| max_accounts += account_preload.maxAccounts(batch.transactions);
+        var accounts = AccountMap{};
+        try accounts.ensureUnusedCapacity(allocator, max_accounts);
+
+        for (batches) |batch| for (batch.transactions) |transaction| {
+            try account_preload.load(
+                &accounts,
                 allocator,
                 params.account_reader,
                 &transaction.accounts,
                 transaction.instructions,
             );
-        }
+        };
```

Comment:
We talked about this before, but I'm not a fan of this because it reduces the pipelining we can do.

Let's pull out the AccountMap soon

Reply:
> Let's pull out the AccountMap soon

done